controller/source/kv_client/rocksdb.hpp

```diff
@@ -1,65 +1,93 @@
 #pragma once
 
-#include <rocksdb/db.h>
-#include <rocksdb/options.h>
+#include <vector>
 
+#include <boost/algorithm/string.hpp>
+#include <boost/asio.hpp>
+
+#include "../rocksdb_server/message.hpp"
 #include "kv_client.hpp"
 
 class rocksdb_client : public kv_client
 {
-rocksdb::DB* m_rocksdb{nullptr};
-
 public:
-  explicit rocksdb_client(const std::string& db_path) {
-    rocksdb::Options options;
-    options.create_if_missing = true;
-    rocksdb::Status status = rocksdb::DB::Open(options, db_path, &m_rocksdb);
-    if (!status.ok()) {
-      std::cerr << "Failed to open database: " << status.ToString() << std::endl;
-    }
+  explicit rocksdb_client(const std::string& addr)
+      : m_socket(m_io_context)
+  {
+    std::vector<std::string> host_port_splits;
+    boost::split(host_port_splits, addr, boost::is_any_of(":"));
+    assert(host_port_splits.size() == 2 && "DB server address must be in <host>:<port> format!");
+
+    // Resolve the host and port to an endpoint
+    using boost::asio::ip::tcp;
```

Comment:
maybe move this using directive on the preamble of the file? would make it cleaner. @ertugrulaypek

Reply:
I put it here because this is the only scope that needs the namespace. I did not want to pollute our namespace. But if that does not make sense to you, I can move it

Reply:
It totally makes sense. Just looked weird compared to other files we have. You can leave it as it is.